src/discof/restore/fd_snapin_tile.c

```diff
@@ -611,6 +636,7 @@ handle_control_frag( fd_snapin_tile_t *  ctx,
         fd_funk_txn_remove_published( ctx->funk );
       } else {
         fd_funk_txn_cancel( ctx->funk, ctx->xid );
```

Comment:
This will deadlock, need to temporarily release record lock before doing txn level operations.